source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,293 @@
+
+All we want to do in our component is make sure its appending the output of the maps service to an inner map container element.
+To limit the test to validating just this behavior we'll take advantage of the registration API to provide a mock maps service.
+In the mock service, define a method that will fetch the map based on location, called `getMapElement`.
+
+```tests/integration/components/location-map-test.js
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+let MockMapsService = Ember.Service.extend({
+  getMapElement(location) {
+    this.set('calledWithLocation', location);
+    // We create a div here to simulate our maps service,
+    // which will create and then cache the map element
+    return document.createElement('div');
+  }
+});
+
+moduleForComponent('location-map', 'Integration | Component | location map', {
+  integration: true,
+  beforeEach() {
+    this.register('service:maps', MockMapsService);
+    this.inject.service('maps', { as: 'mapsService' });
+  }
+});
+
+test('should append map element to container element', function(assert) {
+  this.set('myLocation', 'New York');
+  this.render(hbs`{{location-map location=myLocation}}`);
+  assert.equal(this.$('.map-container').children().length, 1, 'container should have one child');
+  assert.equal(this.get('mapsService.calledWithLocation'), 'New York', 'should call service with New York');
+});
+```
+
+Notice that our mock service above sets a variable called `calledWithLocation`.  We later assert on this variable to ensure that the service is being called with the same location provided to the component.
+
+To get the test to pass, add the container element to the component template.
+
+```app/templates/components/location-map.hbs
+<div class="map-container"></div>
+```
+
+Then update the component to append the map output to our component.
+We'll add a maps service injection, and call the `getMapElement` function with the provided location.
+
+```app/components/location-map.js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  maps: Ember.inject.service(),
+
+  didInsertElement() {
+    this._super(...arguments);
+    let location = this.get('location');
+    let mapElement = this.get('maps').getMapElement(location)
+    this.$('.map-container').append(mapElement);
+  }
+});
+```
+
+### Fetching Maps With a Service
+
+Now we have a passing component integration test, but no maps generating onscreen.
+To generate the maps we'll implement a maps service.
+
+Accessing our maps API through a service will give us several benefits
+
+* It is injected with a [service locator pattern](https://en.wikipedia.org/wiki/Service_locator_pattern), meaning it will abstract the maps API from the code that uses it, allowing for easier refactoring and maintenance
```

Comment:
Can we say this in less 'tech'?  If I'm considering converting to Ember from a jQuery background, my eyes may have just glazed over ;-)  I guess it's worth considering whether simple jQuery users are really our target audience here though ...